data/adapters/gaf_adapter.py

```diff
@@ -0,0 +1,129 @@
+# UniProtKB	A0A024RBG1	NUDT4B	located_in	GO:0005829	GO_REF:0000052	IDA		C	Diphosphoinositol polyphosphate phosphohydrolase NUDT4B	NUDT4B	protein	taxon:9606	20161204	HPA
+# UniProtKB	A0A075B6H7	IGKV3-7	involved_in	GO:0002250	GO_REF:0000043	IEA	UniProtKB-KW:KW-1064	P	Probable non-functional immunoglobulin kappa variable 3-7	IGKV3-7	protein	taxon:9606	20230306	UniProt
+# UniProtKB	A0A075B6H7	IGKV3-7	located_in	GO:0005886	GO_REF:0000044	IEA	UniProtKB-SubCell:SL-0039	C	Probable non-functional immunoglobulin kappa variable 3-7	IGKV3-7	protein	taxon:9606	20230306	UniProt
+
+
+class GAF(Adapter):
+    SKIP_BIOCYPHER = True
+    DATASET = 'gaf'
+    OUTPUT_PATH = './parsed-data'
+    RNACENTRAL_ID_MAPPING_PATH = './samples/rnacentral_ensembl_gencode.tsv.gz'
+    SOURCES = {
+        'human': 'http://geneontology.org/gene-associations/goa_human.gaf.gz',
+        'human_isoform': 'http://geneontology.org/gene-associations/goa_human_isoform.gaf.gz',
+        'rna': 'http://geneontology.org/gene-associations/goa_human_rna.gaf.gz',
+        'rnacentral': 'https://ftp.ebi.ac.uk/pub/databases/RNAcentral/current_release/id_mapping/database_mappings/ensembl_gencode.tsv'
+    }
+
+    def __init__(self, filepath, gaf_type='human', dry_run=True):
+        if gaf_type not in GAF.SOURCES.keys():
+            raise ValueError('Ivalid type. Allowed values: ' +
+                             ', '.join(GAF.SOURCES.keys()))
+
+        self.filepath = filepath
+        self.dataset = GAF.DATASET
+        self.dry_run = dry_run
+        self.type = gaf_type
+        self.output_filepath = '{}/{}-{}.json'.format(
+            GAF.OUTPUT_PATH,
+            self.dataset,
+            filepath.split('/')[-1]
+        )
+
+        super(GAF, self).__init__()
+
+    def load_rnacentral_mapping(self):
+        self.rnacentral_mapping = {}
+        with gzip.open(GAF.RNACENTRAL_ID_MAPPING_PATH, 'rt') as mapping_file:
+            for annotation in mapping_file:
+                mapping = annotation.split('\t')
+                self.rnacentral_mapping[mapping[0] +
+                                        '_' + mapping[3]] = mapping[2]
+
+    def process_file(self):
+        parsed_data_file = open(self.output_filepath, 'w')
+
+        if self.type == 'rna':
+            self.load_rnacentral_mapping()
+
+        with gzip.open(self.filepath, 'rt') as input_file:
+            for annotation in gafiterator(input_file):
+                _from = 'ontology_nodes/obo:' + \
+                    annotation['GO_ID'].replace(':', '_')
+                _to = 'proteins/' + annotation['DB_Object_ID']
+
+                if self.type == 'rna':
+                    transcript_id = self.rnacentral_mapping.get(
+                        annotation['DB_Object_ID'])
+                    if transcript_id is None:
+                        continue
+                    _to = 'transcripts/' + transcript_id
```

Comment:
oh good catch, it should be `ontology node` that we will merge in the ontology PR. It's not a problem for this PR because this adaptor is skipping Biocypher